Assets/Scripts/Block2TextConsoleManager.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+namespace MoveToCode
+{
+
+    public class Block2TextConsoleManager : Singleton<Block2TextConsoleManager>
+    {
+        StandAloneInstruction curInstruction;
+     
+
+        public void UpdateConsoleOnSnap() //every time a block is snapped, the console is updated with code/text
+        {
+            Block2TextConsoleManager.instance.ClearConsole();
+            curInstruction = StartCodeBlock.instance.GetMyInternalIArgument() as StandAloneInstruction;
+
+            while (curInstruction != null){
+                Block2TextConsoleManager.instance.AddLine(curInstruction?.DescriptiveInstructionToString());
+                curInstruction = curInstruction.GetNextInstruction();
+            }
+
+        }
+
+            /* <------copied from ConsoleManager------>*/
```

Comment:
Is there a better way to abstract this then in a C# class?

Reply:
given that the file must be a singleton, I'm not sure how to do it